### PR TITLE
add concrete types to main struct

### PR DESCRIPTION
**Pull request by mrufsvold**
I would love to contribute to JulGame. I thought I'd start with a small change to see if you're interested in more contributions. 

All I did here was look at the uses of every property of `Main` and add a concrete type annotation to its declaration. This should make reading and writing much faster against it.

I also changed the background color to a 3 element Tuple because it doesn't need to change lengths, and you can save allocations and memory indirections by keeping that inline.

I had trouble in two places:
1) There seems to be inconsistency in when the mouse position uses Vector2f `and when it uses `Vector2`. I just set all of the mouse properties to be a union of those. You could get a speed up by picking one and being consistent.
2) ScreenManagement is one of the fields, but it is defined after the `main` global is created so I can't give it a concrete type. This could probably be fixed pretty easily, but I was trying to keep my scope small.

Thank you for this project!!

### CODE REVIEW

File: src/Main.jl [hunk 1]
```diff
@@ -59,7 +60,7 @@ module MainLoop
 			this.selectedEntityIndex = -1
 			this.selectedTextBoxIndex = -1
 			this.selectedEntityUpdated = false
-			this.screenDimensions = C_NULL
+			this.screenDimensions = C_NULL # Why is this a C_NULL by default? why not Math.Vector2(0,0)?
```
Comment (Kyjor) [inline src/Main.jl:63]: You'll see a lot of head-scratchers here. This is my lack of knowledge of Julia along with trying to move quickly. I didn't look, but there may be some logic that checks for a C_NULL before doing anything. If not, we can probably change this to Math.Vector2(0,0)

File: src/Main.jl [hunk 2]
```diff
@@ -47,10 +48,10 @@ module MainLoop
 			this.scene = Scene()
 			this.input = Input()
 
-			this.cameraBackgroundColor = [0,0,0]
+			this.cameraBackgroundColor = (0,0,0)
 			this.close = false
-			this.debugTextBoxes = []
-			this.events = []
+			this.debugTextBoxes = UI.TextBoxModule.TextBox[]
+			this.events = [] # what is this for?
```
Comment (Kyjor) [inline src/Main.jl:54]: This can be removed I believe. I don't think it's used anywhere and is a leftover of old code

File: src/Scene.jl [hunk 3]
```diff
@@ -9,7 +10,7 @@
     function Scene()
         this = new()
 
-        this.camera = C_NULL
+        this.camera = C_NULL # why use C_NULL not nothing?
```
Comment (Kyjor) [inline src/Scene.jl:13]: This is also a lack of knowledge of julia. I think the reason why I use it so much is from using the SDL package. Is using Nothing faster? Definitely need to audit the usage of this

File: src/Main.jl [hunk 4]
```diff
@@ -10,33 +10,34 @@ module MainLoop
...
+		cameraBackgroundColor::Tuple{Int64, Int64, Int64}
 		close::Bool
 		currentTestTime::Float64
-		debugTextBoxes
-		events
-		globals
-		input
+		debugTextBoxes::Vector{UI.TextBoxModule.TextBox}
+		events::Vector # what is this for ?
+		globals::Vector  # what is this for ?
```
Comment (Kyjor) [inline src/Main.jl:18]: This `globals` is a workaround I've created for game projects to use external packages without having to add them to JulGame so I can compile them

**Comment by Kyjor** [comment]
Thank you so much for taking the time to do this!! Sorry for all the inconsistencies in the code. I think that for mouse position, it should just be a `Vector2`.  As for the code coverage issue, don't worry about it. I plan on getting that up to snuff after this next major release. Just let me know when it's ready and I'll merge it. Thanks!!!

File: src/Main.jl (hunk 1 above)
Comment (mrufsvold) [reply src/Main.jl:63]: I'll check before changing anything!

File: src/Scene.jl (hunk 3 above)
Comment (mrufsvold) [reply src/Scene.jl:13]: I don't think it's faster, just idiomatic. I think C_NULL is really just for interfacing with ccalls. Nothing represents "Nothing here for now, I'll fill it in later"

File: src/Main.jl (hunk 4 above)
Comment (mrufsvold) [reply src/Main.jl:18]: Cool!

Comment (mrufsvold) [reply src/Main.jl:18]: Could we call this `Vector{Any}` then?

**Comment by mrufsvold** [comment]
I'll remove my comments and make some final adjustments tomorrow!

File: src/Main.jl (hunk 4 above)
Comment (Kyjor) [reply src/Main.jl:18]: Yeah that'll work!

File: src/Scene.jl (hunk 3 above)
Comment (Kyjor) [reply src/Scene.jl:13]: Gotcha. That might have to be a separate issue for some standardization. I don't know everywhere we check for this value in other files.

**Comment by Kyjor** [comment]
> I'll remove my comments and make some final adjustments tomorrow!

Awesome, thanks!

**Comment by mrufsvold** [comment]
Addressed comments and tests passed locally!

**Comment by mrufsvold** [comment]
I didn't touch the Vector2 stuff. I'll just save that for next time!

**Review by Kyjor — APPROVED** [review]

**Comment by Kyjor** [comment]
Thanks again!

**Pull request merged**